data_steward/cdr_cleaner/cleaning_rules/no_data_30_days_after_death.py

```diff
@@ -0,0 +1,112 @@
+"""
+If there is a death_date listed for a person_id, ensure that no temporal fields
+(see the CDR cleaning spreadsheet tab labeled all temporal here) for that person_id exist more than
+30 days after the death_date.
+"""
+
+import constants.bq_utils as bq_consts
+import constants.cleaners.clean_cdr as cdr_consts
+
+# add table names as keys and temporal representations as values into a dictionary
+TEMPORAL_TABLES_WITH_END_DATES = {'visit_occurrence': 'visit',
+                                  'condition_occurrence': 'condition',
+                                  'drug_exposure': 'drug_exposure',
+                                  'device_exposure': 'device_exposure'}
+
+TEMPORAL_TABLES_WITH_NO_END_DATES = {'person': 'birth',
+                                     'measurement': 'measurement',
+                                     'procedure_occurrence': 'procedure',
+                                     'observation': 'observation',
+                                     'specimen': 'specimen'}
+
+# Join Death to domain_table ON person_id
+# check date field is not more than 30 days after the death date
+# select domain_table_id from the result
+# use the above generated domain_table_ids as a list
+# select rows in a domain_table where the domain_table_ids not in above generated list of ids
+
+REMOVE_DEATH_DATE_QUERY_WITH_END_DATES = (
+    "SELECT * "
+    "FROM `{dataset}.{table_name}` "
+    "WHERE {table_name}_id NOT IN ("
+    "  SELECT ma.{table_name}_id "
+    "  FROM `{dataset}.{table_name}` ma"
+    "  JOIN `{dataset}.death` d"
+    "  ON ma.person_id = d.person_id"
+    "  WHERE (date_diff(ma.{start_date},d.death_date, DAY) > 30"
```

Comment:
This `WHERE` clause is my only other concern.  The issue says, 

> ensure that no records for that person_id exist more than 30 days after the death date.

This clause will allow records with a start_date of 29 or less days and an end_date of 30 or more days to remain in the CDR.  Is this desired?  Should the `AND` in the clause become an `OR`?  Allowing if either start or end date time is more than 30 past the death date, the entire row should be identified and dropped.